implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -191,13 +191,75 @@
         return getValue(name, requireConverter(aClass));
     }
 
+    /**
+     * Return the content of the direct sub properties as the requested type of Map.
+     *
+     * @param name The configuration property name
+     * @param kClass the type into which the keys should be converted
+     * @param vClass the type into which the values should be converted
+     * @param <K> the key type
+     * @param <V> the value type
+     * @return the resolved property value as an instance of the requested Map (not {@code null})
+     * @throws IllegalArgumentException if a key or a value cannot be converted to the specified types
+     * @throws NoSuchElementException if no direct sub properties could be found.
+     */
+    @Experimental("Extension to retrieve mandatory sub properties as a Map")
+    public <K, V> Map<K, V> getValues(String name, Class<K> kClass, Class<V> vClass) {
+        final Map<K, V> result = valuesAsMap(name, requireConverter(kClass), requireConverter(vClass));
+        if (result == null) {
+            throw new NoSuchElementException(ConfigMessages.msg.propertyNotFound(name));
+        }
+        return result;
+    }
+
+    /**
+     * Return the content of the direct sub properties as the requested type of Map.
+     *
+     * @param name The configuration property name
+     * @param keyConverter The converter to use for the keys.
+     * @param valueConverter The converter to use for the values.
+     * @param <K> The type of the keys.
+     * @param <V> The type of the values.
+     * @return the resolved property value as an instance of the requested Map or {@code null} if it could not be found.
+     * @throws IllegalArgumentException if a key or a value cannot be converted to the specified types
+     */
+    public <K, V> Map<K, V> valuesAsMap(String name, Converter<K> keyConverter, Converter<V> valueConverter) {
```

Comment:
I could not use getValues as name here because it is in conflict with `getValues(String name, Converter<T> converter, IntFunction<C> collectionFactory)`

Reply:
Ok, let's just use `getValuesAsMap`. So we can keep the `get` consistency.